DataFrameClient for InfluxDB 0.9

tzonghao opened the pull request:
#108 

* `get_list_series` returns a mapping from measurement names to DataFrame representing tags for the named measurement.
* `query` returns a mapping from (measurement, (tag (k, v) pairs), to DataFrame representing time series for the measurement (qualified by tags) in question. Query without tags should simply use measurement as keys.
* `write_points` converts a mapping of DataFrame objects as `points`, then hand them over to super's `write_points`.

ReAzem commented:
Hi,

Thank you for taking on this issue :)

Have you looked at ``client_test_with_server.py``? I would like some DaframeClient tests there.

You are using the ``u'string'`` syntax, which does not work in python 3.2.

Other than that, this looks great!

//cc @timtroendle

tzonghao commented:
@ReAzem In case you haven't reviewed it, a few basic tests were added in `client_test_with_server.py` in this pull request as suggested. The 3.2 syntax problem is also fixed. I've also changed the key data structure for data dictionary's keys, for both querying and submitting, where strings can be used as keys if no tags are used. Otherwise, the keys can be a tuple of (measurement, sorted tuple of tag key-value pairs).

ReAzem commented:
@tzonghao Hello,

Than you for the great work. Why are you not using the ResultSet Object? You should not have to use the raw property.

tzonghao commented:
@ReAzem You're right. Using ResultSet is cleaner. It's converted in the last commit.

ReAzem commented:
@tzonghao Can you change the write_points function to accept three arguments instead?

- measurment name
- tags (optional)
- dataframe

I think it would be more intuitive than a tuple of key-value tuples.

tzonghao commented:
@ReAzem Sure, I can make that change. It also implicates that `write_points` will accept one DataFrame at a time, instead of multiple DataFrames.

ReAzem commented:
@tzonghao  I think it would be worth it, do you agree?

tzonghao commented:
I agree. It's a good decision.

tzonghao commented:
@ReAzem It looks like an InfluxDB error in the last Travis run that failed the py34 test. Can you trigger a rerun?

ReAzem commented on examples/tutorial_pandas.py:
```diff
@@ -22,6 +22,9 @@ def main(host='localhost', port=8086):
     client.write_points({'demo': df})
 
+    print("Write DataFrame with Tags")
+    client.write_points({('demo', (('k1', 'v1'), ('k2', 'v2'))): df})
```
I don't think this works anymore

tzonghao replied:
good catch

ReAzem commented:
Great work! I will be merging this for now. I'm sure we can keep on improving it after feedback from other users.
